pandas/core/groupby/ops.py

```diff
@@ -680,8 +740,13 @@ def _cython_operation(
         assert axis == 1
         values = values.T
 
+        if mask is not None:
+            mask = mask.reshape(values.shape, order="C")
+
         out_shape = cy_op.get_output_shape(ngroups, values)
         func, values = cy_op.get_cython_func_and_vals(values, is_numeric)
+        if func_uses_mask:
+            func = partial(func, mask=mask)
```

Comment:
is there a viable alternative to using a partial here?  When its just one its not that bad, but they have a tendency to pile up and make the code harder to reason about

Reply:
Makes sense. Will look into it

Reply:
Called `func` with if/else structure instead of using partial, which is definitely easier to follow if a little longer (but aggregate does something similar).